qcodes/instrument/base.py

```diff
@@ -58,8 +58,8 @@ def __init__(self, name: str,
 
         # This is needed for snapshot method to work
         self._meta_attrs = ['name']
-
-        self.log = get_instrument_logger(self, __name__)
+        log_name = ".".join(self._logger_name_parts)
```

Comment:
so this delimiter is to connect parent instrument and the channel, right? then i'd suggest to not use `.` as delimiter becuase it may get confused with the module/classname.
```suggestion
        log_name = "--".join(self._logger_name_parts)
```

Reply:
I don't think that will work. We explicit want the channel logger to be a child of the instrument logger. As I understand it that depends on using `.` as a separator

Reply:
But I will check

Reply:
A quick manual test verified that this is indeed the case. To establish the child-parent relationship the names must be separated by `.`. This should be incorporated into the tests

Reply:
This is unfortunate :( but great to know and yes , shouldbe tested for :)